lib/pages/farm/usecases/get_farm_data_use_case.dart

```diff
@@ -18,40 +21,18 @@ class GetFarmDataUseCase {
       throw response.exception.toString();
     }
     return _mapQueryResultToFarmModel(
-      response.data!['user']!['positions'],
+      response.data!['user']!['positions'] as List<dynamic>,
```

Comment:
You're banging a lot of fields there, are you sure they can never be null and result in an error ?